File: go/robot-name/robot_name.go
package robotname

import (
	"errors"
	"math/rand"
	"strings"
	"time"
)

type Robot struct {
	name string
}

const max = 26 * 26 * 10 * 10 * 10

var used = map[string]bool{}

func (r *Robot) Name() (string, error) {

	if r.name == "" {

		if len(used) == max {
			return "", errors.New("no more names")
		}

		r.Reset()
	}

	return r.name, nil

}

func (r *Robot) Reset() {

	for {
		var name = gen()
		if !used[name] {
			r.name = name
			break
		}

	}

}

func gen() string {

	s1 := rand.NewSource(time.Now().UnixNano())
	r1 := rand.New(s1)

	var name strings.Builder

	var c1 = rune(r1.Intn(26) + 'A')
	var c2 = rune(r1.Intn(26) + 'A')

	var i1 = rune(r1.Intn(10) + '0')
	var i2 = rune(r1.Intn(10) + '0')
	var i3 = rune(r1.Intn(10) + '0')

	name.WriteRune(c1)
	name.WriteRune(c2)
	name.WriteRune(i1)
	name.WriteRune(i2)
	name.WriteRune(i3)

	return name.String()

}

File: php/robot-name/RobotName.php
<?php

class Robot
{
    public string $name = "";
    public static array $names = [];

    public function __construct()
    {
        $this->reset();
    }

    public function getName(): string
    {
        return $this->name;
    }

    public function reset(): void
    {
        $letters = [];
        for ($i=0; $i<5; $i++) {
            if ($i < 2)
                array_push($letters, chr(rand(65,90)));
            else
                array_push($letters, rand(0,9));
        }
        $cand = implode("", $letters);

        if (in_array($cand, self::$names)) {
            $this->reset();
        } else {
            array_push(self::$names, $cand);
            $this->name = $cand;
        }
    }
}

